imageio/imageio-tiff/src/main/java/com/twelvemonkeys/imageio/plugins/tiff/CCITTFaxDecoderStream.java

```diff
@@ -150,15 +150,21 @@ public CCITTFaxDecoderStream(final InputStream stream, final int columns, final
     }
 
     static int findCompressionType(final int encodedType, final InputStream stream) throws IOException {
+        return findCompressionType(encodedType, stream, TIFFBaseline.FILL_LEFT_TO_RIGHT);
+    }
+
+    static int findCompressionType(final int encodedType, final InputStream stream, int fillOrder) throws IOException {
         // Discover possible incorrect compression type, revert to RLE if no EOLs found
         if (encodedType == TIFFExtension.COMPRESSION_CCITT_T4 && stream.markSupported()) {
             int limit = 512;
-
             try {
                 stream.mark(limit);
-
-                int first = stream.read();
-                int second = stream.read();
+                byte first = (byte) stream.read();
+                byte second = (byte) stream.read();
```

Comment:
I don't think it's safe to cast these values to `byte` right away, as you'll mask the `-1`/EOF value. The test for `-1` later on will also potentially match for `0xFF`...